src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -67,6 +80,50 @@ public FilterPanel(UI ui, IModel model, PanelControl parentPanelControl, int pan
 
         filterTextField.setKeywords(all);
     };
+    
+    private Node createNameBar() {
+        nameText = new Text(panelName);
+    
+        renameButton = new Label(RENAME_PANEL);
+        renameButton.getStyleClass().add("label-button");
+        renameButton.setId(model.getDefaultRepo() + "_col" + panelIndex + "_renameButton");
+        renameButton.setOnMouseClicked(e -> {
+            TextInputDialog renameDialog = new TextInputDialog(panelName);
+            renameDialog.getEditor().setId("panelrenameinput");
+            renameDialog.setTitle("Rename " + panelName);
+            renameDialog.setHeaderText("Enter a new name for this panel.");
+            Optional<String> result = renameDialog.showAndWait();
+            mainStage.show();
```

Comment:
It is used with functions that bring out new dialogs, like ```LabelPicker``` because Travis fails to go back to the main stage after it's done with the dialog. I'll add comments to indicate this.

If events are used, I can move the event handler and ```mainStage.show()``` to ```PanelControl```. Is this okay in terms of coupling?